test/unit/raft.c

```diff
@@ -1893,10 +1893,277 @@ raft_test_split_vote(void)
+		1 /* Leader. */,
+		5 /* Term. */,
+		1 /* Vote. */,
+		5 /* Volatile term. */,
+		1 /* Volatile vote. */,
+		"{0: 5}" /* Vclock. */
+	), "Term bump with cfg_is_candidate_later transfers voter to candiate");
+
+	is(raft_leader_idle(&node.raft), 0,
+	   "leader_idle is zero on the current leader");
+
+	raft_node_cfg_is_candidate(&node, false);
+
+	raft_run_for(node.cfg_death_timeout / 2);
+	is(raft_leader_idle(&node.raft), node.cfg_death_timeout / 2,
+	   "leader_idle counts from 0 on a previous leader")
+
+	raft_node_cfg_is_enabled(&node, false);
+
+	is(raft_node_send_is_leader_seen(
+		&node,
+		6 /* Term. */,
+		true /* Is leader seen. */,
+		2 /* Source. */
+	), 0, "leader is seen message accepted when raft disabled");
+
+	ok(&node.raft.leader_witness_map != 0,
+	   "who sees leader is tracked on disabled node");
+
+	ok(!raft_ev_is_active(&node.raft.timer),
+	   "disabled node doesn't wait for anything");
+
+	raft_node_cfg_is_candidate(&node, true);
+	raft_node_cfg_is_enabled(&node, true);
+
+	ok(raft_node_check_full_state(
+		&node,
+		RAFT_STATE_FOLLOWER /* State. */,
+		0 /* Leader. */,
+		6 /* Term. */,
+		0 /* Vote. */,
+		6 /* Volatile term. */,
+		0 /* Volatile vote. */,
+		"{0: 6}" /* Vclock. */
+	), "no elections on start when someone sees the leader");
+
+	ok(!raft_ev_is_active(&node.raft.timer),
+	   "nothing to wait for as long as someone sees the leader");
+
+	raft_node_cfg_is_candidate(&node, false);
+	raft_node_cfg_is_candidate(&node, true);
+
+	ok(raft_node_check_full_state(
+		&node,
+		RAFT_STATE_FOLLOWER /* State. */,
+		0 /* Leader. */,
+		6 /* Term. */,
+		0 /* Vote. */,
+		6 /* Volatile term. */,
+		0 /* Volatile vote. */,
+		"{0: 6}" /* Vclock. */
+	), "no elections on becoming candidate when someone sees the leader");
+
+	ok(!raft_ev_is_active(&node.raft.timer),
+	   "nothing to wait for as long as someone sees the leader");
+
+	is(raft_node_send_leader(
+		&node,
+		6 /* Term. */,
+		3 /* Source. */
+	), 0, "leader is accepted");
+
+	raft_run_for(node.cfg_death_timeout * 2);
+	ok(!raft_ev_is_active(&node.raft.timer), "timed out");
+	raft_node_cfg_death_timeout(&node, node.cfg_death_timeout / 2);
+	ok(!raft_ev_is_active(&node.raft.timer),
+	   "No timer re-start on death timeout reconfig "
+	   "when already timed-out");
+
+	is(raft_node_send_vote_request(
+		&node,
+		7 /* Term. */,
+		"{2: 1}" /* Vclock. */,
+		2 /* Source. */
+	), 0, "vote for 2");
+	is(raft_node_send_leader(
+		&node,
+		7 /* Term. */,
+		2 /* Source. */
+	), 0, "leader accepted");
+	is(raft_node_send_is_leader_seen(
+		&node,
+		7 /* Term. */,
+		true /* Is leader seen. */,
+		3 /* Source. */
+	), 0, "leader seen notification accepted");
+
+	raft_run_for(node.cfg_death_timeout * 2);
+	raft_node_cfg_election_timeout(&node, node.cfg_election_timeout / 2);
+	ok(!raft_ev_is_active(&node.raft.timer),
+	   "No timer re_start on election timeout reconfig "
+	   "when it's not time for elections yet");
+
+	raft_node_destroy(&node);
+	raft_finish_test();
```

Comment:
See a not resolved comment from the previous review about a test for leader heartbeats during WAL write.

Reply:
I missed that, sorry! Applied.